### ISSUE TITLE
Add GitHub Oauth registration and login

### ISSUE THREAD

**Issue by isabelcosta** [anitab-org/open-source-programs-backend#101]
## Description

As a developer,
I need to implement GitHub sign on,
so that I can allow GitHub users to login using their GitHub account.

For being able to use the application, the user has to create a new account and confirm the email just to login. It's a long process. So by having GitHub OAuth we can make the process shorter for users who have a GitHub account.

### Update [Required]
- [ ] Implement GitHub OAuth login and registration
- [ ] Write unit tests for the updates ^^

## Definition of Done
- [ ] All of the required items are completed.
- [x] Approval by 1 mentor.

## Estimation
8 hours

**Comment by jalajcodes** [comment]
Can I work on it? @isabelcosta

**Comment by codesankalp** [comment]
@jalajcodes You are assigned to https://github.com/anitab-org/open-source-programs-backend/issues/63.

@isabelcosta I want to give it a try.

**Comment by isabelcosta** [comment]
assigned to @codesankalp because you're not assigned to anything

**Comment by codesankalp** [comment]
@isabelcosta Can I create an issue on the frontend also for this and start working on both as I will need the frontend to test the implementation from the backend?

**Comment by isabelcosta** [comment]
Yes, you can do that and build the feature full stack. Do create the issue and link to this comment of mine and to this issue. @codesankalp

**Comment by codesankalp** [comment]
**Update:** Implemented the Github OAuth on both frontend and backend but facing problem in making jwt authorization. Will solve this problem and make a PR asap.

**Comment by isabelcosta** [comment]
Awesome @codesankalp I love me some updates 💯 thank you!
